src/Components/InputDate.php

```diff
@@ -0,0 +1,171 @@
+ *
+ * It is a visual component for InputDate, an EditBox to hold a date, with an attached SpeedButton that will summon a date selection (calendar) dialog.
+ *
+ * @author Everton da Rosa everton3x@gmail.com
+ * @since 0.1
+ */
+class InputDate extends VisualObject
+{
+
+    /**
+     * dsShowHeadings
+     */
+    const SHOW_HEADINGS = 'dsShowHeadings';
+
+    /**
+     * dsShowDayNames
+     */
+    const SHOW_DAY_NAMES = 'dsShowDayNames';
+
+    /**
+     * dsNoMonthChange
+     */
+    const NO_MONTH_CHANGE = 'dsNoMonthChange';
+
+    /**
+     * dsShowWeekNumbers
+     */
+    const SHOW_WEEK_NUMBERS = 'dsShowWeekNumbers';
+
+    /**
+     * dsStartMonday
+     */
+    const START_MONDAY = 'dsStartMonday';
+
+    /**
+     * The lazarus class as string
+     *
+     * @var string $lazarusClass
+     */
+    protected $lazarusClass = 'TDateEdit';
+
+    /**
+     * Sets if the button only appears when the control receives focus.
+     *
+     * @param bool $option True to activate option. False is default.
+     * @return self
+     */
+    public function setButtonOnlyWhenFocused($option)
+    {
+        $this->set('ButtonOnlyWhenFocused', $option);
+
+        return $this;
+    }
+
+    /**
+     * Gets if the button only appears when the control receives focus.
+     *
+     * @return bool
+     */
+    public function isButtonOnlyWhenFocused()
+    {
+        return (bool) $this->get('ButtonOnlyWhenFocused');
+    }
+
+    /**
+     * Sets if direct data input to the Edit Box is permitted.
+     *
+     * @param string $bool  if True, direct data input to the Edit Box is permitted. Default is TRUE.
+     *
+     * @return self
+     */
+    public function setAcceptInput($bool)
+    {
+        $this->set('DirectInput', $bool);
+
+        return $this;
+    }
+
+    /**
+     * Gets if direct data input to the Edit Box is permitted.
+     *
+     *
+     * @return string
+     */
+    public function getAcceptInput()
+    {
+        return $this->get('DirectInput');
+    }
+
+    /**
+     * Sets the value of component.
+     *
+     * @param string $value the value
+     *
+     * @return self
+     */
+    public function setValue($value)
+    {
+        $this->set('Text', $value);
+
+        return $this;
+    }
+
+    /**
+     * Gets the value of component;
+     *
+     * @return string
+     */
+    public function getValue()
+    {
+        return $this->get('Text');
+    }
+
+    /**
+     * Returns a string with the representation [option1, option2, ...] to be configured by Lazarus.
+     *
+     * @param array $options the options
+     * @return string a string [option1, options2, option3,...]
+     */
+    private function getOptionString($options)
+    {
+        $str = join(', ', $options);
+        return "[$str]";
+    }
+
+    /**
+     * Parse an option string received from Lazarus.
+     *
+     * @param string $options the options string
+     * @return array
+     */
+    private function parseOptionString($options)
```

Comment:
Would it be convenient to move it to a trait, or an intermediate class that extends VisualObject to be then extended by InputDate and InputFile?